src/core_plugins/kibana/public/dashboard/dashboard_state_manager.js

```diff
@@ -204,7 +204,7 @@ export class DashboardStateManager {
       FilterUtils.cleanFiltersForComparison(this.appState.filters),
       FilterUtils.cleanFiltersForComparison(getFilters(state))
     )) {
-      store.dispatch(updateFilters(this.appState.filters));
+      store.dispatch(updateFilters(_.cloneDeep(this.appState.filters)));
```

Comment:
If this fixes the problem, it indicates that some code is mutating `appState.filters`, which seems like the real source of the bug, but also probably too tricky to fix right now. Given that redux expects its state to be immutable, I think we need to change where this fix goes by putting the clone deep in the action helper (`updateFilters`). Putting the cloneDeep in the action helper means that any other code that dispatches an updateFilters action will not be victim to the same bug. Does that make sense?